pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -115,3 +200,38 @@ func getInfraMachineFromProvider(platform configv1.PlatformType) (client.Object,
 		return nil, fmt.Errorf("%w: %s", errPlatformNotSupported, platform)
 	}
 }
+
+// shouldMirrorCAPIMachineToMAPIMachine takes a CAPI machine and determines if there should
+// be a MAPI mirror, it returns true only if:
+//
+// 1. The CAPI Machine is owned by a CAPI MachineSet,
+// 2. That CAPI MachineSet has a MAPI MachineSet Mirror.
+func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context.Context, machine *capiv1beta1.Machine, logger logr.Logger) (bool, error) {
+	// Check if owner refs point to CAPI MachineSet
+	for _, ref := range machine.ObjectMeta.OwnerReferences {
+		if ref.Kind != "MachineSet" || ref.APIVersion != "cluster.x-k8s.io/v1beta1" {
+			continue
+		}
+
+		// Checks if the CAPI MS has a mirror in MAPI namespace
+		key := client.ObjectKey{
+			Namespace: r.MAPINamespace,
+			Name:      ref.Name,
+		}
+		mapiMachineSet := &machinev1beta1.MachineSet{}
+
+		if err := r.Get(ctx, key, mapiMachineSet); apierrors.IsNotFound(err) {
+			logger.Info("MAPI MachineSet mirror not found, nothing to do", ref.Name)
+
+			return false, nil
+		} else if err != nil {
+			logger.Error(err, "Failed to get MAPI MachineSet mirror")
+
+			return false, fmt.Errorf("%w: %w", errFailedToGetMAPIMachineSet, err)
+		}
+
+		return true, nil
+	}
+
+	return false, nil
```

Comment:
What happens if the CAPI machine has no owner references at all? I think this is correct, that we don't reconcile, but have we double checked/should we add a comment?